IFTPipeline.jl/src/feature-extraction.jl

```diff
@@ -48,16 +48,21 @@ julia> IFTPipeline.extractfeatures(bw_img; minarea=minarea, maxarea=maxarea, fea
 ```
 """
 function extractfeatures(
-    bw::T;
+    floes::AbstractArray{<:Union{Integer,Bool}};
     minarea::Int64=350,
     maxarea::Int64=90000,
     features::Union{Vector{Symbol},Vector{<:AbstractString}}
-)::DataFrame where {T<:AbstractArray{Bool}}
+)::DataFrame
     # assert the first area threshold is less than the second
     minarea >= maxarea &&
         throw(ArgumentError("The minimum area must be less than the maximum area."))
 
-    props = regionprops_table(label_components(bw, trues(3, 3)); properties=features)
+    if eltype(floes) == Bool
+        props = regionprops_table(label_components(floes, trues(3, 3)); properties=features)
+    else
+        props = regionprops_table(floes; properties=features)
+    end
```

Comment:
Use ternary operator to update the `floes` object and call `props = regionprops_table(floes; properties=features)` afterwards.
```suggestion
    floes = isa(eltype(floes), Bool) ? label_components(floes, trues(3, 3)) : floes
    props = regionprops_table(floes; properties=features)
```

Reply:
I like it!